crypto/s2n_ecdsa.c

```diff
@@ -157,26 +161,26 @@ int s2n_ecdsa_signature_size(const struct s2n_ecdsa_private_key *key)
     return ECDSA_size(key->eckey);
 }
 
-int s2n_ecdsa_keys_match(const struct s2n_ecdsa_public_key *pub_key, const struct s2n_ecdsa_private_key *priv_key) 
+int s2n_ecdsa_keys_match(const struct s2n_ecdsa_public_key *pub_key, struct s2n_hash_state *state_in,
+                         const struct s2n_ecdsa_private_key *priv_key, struct s2n_hash_state *state_out)
```

Comment:
I agree. Yes, this change was intended to avoid any request path allocation of s2n_hash_states (as you can see in the ECDSA unit test where they are allocated before this call). Given your work in this ECDSA area, do you have any thoughts on how to avoid this allocation?